app/models/discourse_post_event/event.rb

```diff
@@ -286,12 +286,34 @@ def self.update_from_raw(post)
       end
     end
 
-    def missing_users(excluded_ids = self.invitees.select(:user_id))
-      User
-        .joins(:groups)
-        .where('groups.name' => self.raw_invitees)
-        .where.not(id: excluded_ids)
-        .distinct
+    def missing_user_ids(excluded_ids = self.invitees.pluck(:user_id))
+      missing_users_sql_query = <<-SQL
+        SELECT "group_users"."user_id"
+        FROM "group_users"
+        INNER JOIN "groups"
+          ON "groups"."id" = "group_users"."group_id"
+      SQL
+
+      if self.raw_invitees.present?
+        missing_users_sql_query += <<-SQL
+          AND "groups"."name" IN ('#{self.raw_invitees.join("', '")}')
```

Comment:
This is a little scary to me. The combination of the word `raw` with interpolated SQL definitely raises an eyebrow for me.

Does mini sql not allow us to pass an array as a sanitized parameter?

Reply:
Yes I'm not happy of this yet, will rework on it